Shell/ConditionalRedundancyHandler.cpp

```diff
@@ -606,4 +605,43 @@ bool ConditionalRedundancyHandlerImpl<enabled, ordC, avatarC, litC>::handleReduc
   return true;
 }
 
+/**
+ * This function is similar to @b DemodulationHelper::isPremiseRedundant.
+ * However, here we do not assume that the rewriting equation is unit, which necessitates some additional checks.
+ */
+template<bool enabled, bool ordC, bool avatarC, bool litC>
+bool ConditionalRedundancyHandlerImpl<enabled, ordC, avatarC, litC>::isSuperpositionPremiseRedundant(
+  Clause* rwCl, Literal* rwLit, TermList rwTerm, TermList tgtTerm, Clause* eqCl, TermList eqLHS,
+  const SubstApplicator* eqApplicator, Ordering::Result& tord) const
```

Comment:
This seems like it should produce linker errors. Why doesn't it?

Reply:
What do you mean? :smile:

Reply:
So there's this template function at the end of the file with 2^4 possible instances (I think), but it doesn't seem to be explicitly instantiated. I think I figured it out: you call the instances you need around line 303, right?

Reply:
Yep, on line 442 in particular. I'm guessing the explicit instantiations in the factory function above causes a cascade of instantiations of everything needed. 🪄